models/migrations/v1_21/v276.go

```diff
@@ -126,6 +129,8 @@ func migratePushMirrors(x *xorm.Engine) error {
 		var mirrors []PushMirror
 		if err := sess.Select("push_mirror.id, push_mirror.repo_id, push_mirror.remote_name, push_mirror.remote_address, repository.owner_name as repo_owner, repository.name as repo_name").
 			Join("INNER", "repository", "repository.id = push_mirror.repo_id").
+			Where("repository.status = ?", repo_model.RepositoryReady).
+			And("repository.is_empty = ?", false).
```

Comment:
An empty repository should have no any git data. So this will avoid read git data here failure. Maybe we can update the remote_address when the first mirror successfully?